tests/contrib/io/azure/test_csv_blob.py

```diff
@@ -62,6 +74,103 @@ def make_data_set(load_args=None, save_args=None):
     return make_data_set
 
 
+@pytest.fixture
+def versioned_blob_csv_data_set(load_version, save_version):
+    return CSVBlobDataSet(
+        filepath=TEST_FILE_NAME,
+        container_name=TEST_CONTAINER_NAME,
+        credentials=TEST_CREDENTIALS,
+        blob_to_text_args={"to_extra": 42},
+        blob_from_text_args={"from_extra": 42},
+        version=Version(load_version, save_version),
+    )
+
+
+@pytest.fixture
+def save_path(save_version):
+    return "{0}/{1}/{0}".format(TEST_FILE_NAME, save_version)
+
+
+class TestCSVBlobDataSetVersioned:
+    # pylint: disable=too-many-arguments
+    @patch("kedro.contrib.io.azure.csv_blob.BlockBlobService.create_blob_from_text")
+    @patch(
+        "kedro.contrib.io.azure.csv_blob.BlockBlobService.exists", return_value=False
+    )
+    def test_save(
+        self,
+        exists_mock,
+        save_mock,
+        versioned_blob_csv_data_set,
+        dummy_dataframe,
+        save_path,
+    ):
+        """Test that saving saves with a correct version"""
+        versioned_blob_csv_data_set.save(dummy_dataframe)
+        save_mock.assert_called_with(
+            container_name=TEST_CONTAINER_NAME,
+            blob_name=save_path,
+            text=dummy_dataframe.to_csv(index=False),
+            from_extra=42,
+        )
+
+    @patch(
+        "kedro.contrib.io.azure.csv_blob.BlockBlobService.list_blob_names",
+        return_value=list(),
+    )
+    @patch(
+        "kedro.contrib.io.azure.csv_blob.BlockBlobService.exists", return_value=False
+    )
+    def test_no_versions(self, exists_mock, list_mock, versioned_blob_csv_data_set):
+        """Check the error if no versions are available for load."""
+        pattern = r"Did not find any versions for CSVBlobDataSet\(.+\)"
+        with pytest.raises(DataSetError, match=pattern):
+            versioned_blob_csv_data_set.load()
+
+    # pylint: disable=too-many-arguments
+    @patch("kedro.contrib.io.azure.csv_blob.BlockBlobService.create_blob_from_text")
+    @patch(
+        "kedro.contrib.io.azure.csv_blob.BlockBlobService.exists", return_value=False
+    )
+    @patch("kedro.contrib.io.azure.csv_blob.CSVBlobDataSet._get_load_path")
+    def test_exists(
+        self,
+        load_mock,
+        exists_mock,
+        save_mock,
+        versioned_blob_csv_data_set,
+        dummy_dataframe,
+        save_path,
+    ):
+        versioned_blob_csv_data_set.save(dummy_dataframe)
+        load_mock.return_value = PurePath(save_path)
+        versioned_blob_csv_data_set.exists()
+        exists_mock.assert_called_with(TEST_CONTAINER_NAME, blob_name=save_path)
+
```

Comment:
Could you also add `test_prevent_override` and `test_save_version_warning` (see the unit tests for other versioned datasets)?

Reply:
Done!